src/tests/krate/publish/oidc.rs

```diff
@@ -0,0 +1,128 @@
+use crate::tests::builders::PublishBuilder;
+use crate::tests::util::oidc::GitHubClaims;
+use crate::tests::util::{MockTokenUser, RequestHelper, TestApp};
+use crates_io_github::{GitHubRepository, GitHubRepositoryOwner, MockGitHubClient};
+use crates_io_oidc::github::GITHUB_ISSUER_URL;
+use crates_io_oidc::keystore::MockOidcKeyStore;
+use crates_io_oidc::test_keys::encode_for_testing;
+use http::StatusCode;
+use insta::assert_json_snapshot;
+use mockall::predicate::*;
+use serde_json::json;
+
+/// Test the full flow of publishing a crate with OIDC authentication
+/// (aka. "Trusted Publishing")
+///
+/// This test will:
+///
+/// 1. Publish a new crate via API token.
+/// 2. Create a Trusted Publishing configuration.
+/// 3. Generate a new OIDC token and exchange it for a temporary access token.
+/// 4. Publish a new version of the crate using the temporary access token.
+#[tokio::test(flavor = "multi_thread")]
+async fn test_full_flow() -> anyhow::Result<()> {
```

Comment:
if you want to start your review somewhere, this is probably the right place for a high level overview :)